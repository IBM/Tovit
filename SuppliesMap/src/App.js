import React, { Component } from 'react'
import './App.css'

import axios from 'axios'

class App extends Component {


  state = {
    venues: []
  }

  componentDidMount() {
    this.getVenues()
  }

  renderMap = () => {
    loadScript("https://maps.googleapis.com/maps/api/js?key=<<GOOGLE MAP API KEY>>&callback=initMap")  // Enter your Google Map API Key here
    window.initMap = this.initMap
  }

  getVenues = () => {
    const endPoint = "https://api.foursquare.com/v2/venues/explore?"
    const parameters = {
      client_id: "PMHC2WA1VCBHVYOPPSJ0QSBYTLRF4PNJ04OWVWV0PZJ0QFIR",
      client_secret: "CULSZZ44YAEBOWBFGPB4BF5ISRXXSNYR0EE3JV3CNE2ZWHV0",
      query: "groceries",
      near: "Dubai",
      v: "20182507"
    }

    axios.get(endPoint + new URLSearchParams(parameters))
      .then(response => {
        this.setState({
          venues: response.data.response.groups[0].items
        }, this.renderMap())
      })
      .catch(error => {
        console.log("ERROR!! " + error)
      })

  }

  initMap = () => {

    // Create A Map
    var map = new window.google.maps.Map(document.getElementById('map'), {
      center: {lat: 25.0769, lng: 55.1341},
      zoom: 8
    })

    // Create An InfoWindow
    var infowindow = new window.google.maps.InfoWindow()

    // Display Dynamic Markers
    this.state.venues.map(myVenue => {

      var contentString = `${myVenue.venue.name}`

      // Create A Marker
      var marker = new window.google.maps.Marker({
        position: {lat: myVenue.venue.location.lat , lng: myVenue.venue.location.lng},
        map: map,
        title: myVenue.venue.name
      })

      // Click on A Marker!
      marker.addListener('click', function() {

        // Change the content
        infowindow.setContent(contentString)

        // Open An InfoWindow
        infowindow.open(map, marker)
      })

    })

    

  }

  render() {
    return (
      <main>
        <div id="map"></div>
      </main>
    )
  }
}

function loadScript(url) {
  var index  = window.document.getElementsByTagName("script")[0]
  var script = window.document.createElement("script")
  script.src = url
  script.async = true
  script.defer = true
  index.parentNode.insertBefore(script, index)
}

export default App;
