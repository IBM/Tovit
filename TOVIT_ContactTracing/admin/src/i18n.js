import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import en from './translations/en'

// the translations
// (tip move them in a JSON file and import them)
const resources = {
  en
}

i18n
  .use(LanguageDetector)
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    resources,
    fallbackLng: 'en',
    keySeparator: false, // we do not use keys in form messages.welcome
    interpolation: {
      escapeValue: false // react already safes from xss
    },
    detection: {
      order: ['querystring', 'cookie', 'localStorage', 'navigator', 'htmlTag', 'path', 'subdomain'],

      // keys or params to lookup language from
      lookupQuerystring: 'lng',
      lookupCookie: 'i18next',
      lookupLocalStorage: 'i18nextLng',
      lookupFromPathIndex: 0,
      lookupFromSubdomainIndex: 0,

      // cache user language on
      caches: ['localStorage', 'cookie'],
      excludeCacheFor: ['cimode'], // languages to not persist (cookie, localStorage)

      // optional expire and domain for set cookie
      cookieMinutes: 10,
      cookieDomain: 'myDomain',

      // optional htmlTag with lang attribute, the default is:
      htmlTag: document.documentElement,

      // only detect languages that are in the whitelist
      checkWhitelist: true
    }
  })

export default i18n
