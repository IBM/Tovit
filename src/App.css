html {
  box-sizing: border-box;
  /* background-color:rgb(194, 196, 255); */
  /* background-color: rgb(0, 176, 185); */
  /* background-color: #004aff; */
  background-color: rgb(44, 74, 219);
  background-repeat: no-repeat;
  background-position: middle;
  background-size: 100%;
  opacity: 0.8;
}

.background-image {
  opacity: 0.2;
  height: 100%;
  width: 100%;
  
}

*, *::before, *::after {
  box-sizing: inherit;
  padding: 0;
  margin: 0;
}

.app {
  width: 960px;
  margin: 0 auto;
  padding: 20px;
}

nav ul {
  list-style: inside;
  display: flex;
  /* background-color: black; */
  margin-bottom: 20px;
}

nav ul li {
  padding: 20px;
  width: 40%;
}

nav ul li a {
  color: white;
  text-decoration: none;
  font-weight: bold;
  
}

.current {
  border-bottom: 4px solid white;
}

h1 {
  margin-bottom: 20px;
  text-align: center;
  /* color: white; */
}

p {
  margin-bottom: 15px;
  /* color: white; */
}

.covid-data-table-wrapper {
  overflow-y: auto;
  height: 350px;
}

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}


img {
  width: 40px;
  height: 40px;
  margin-top: 8px;
  margin: auto 5px;
  float: left;
}

.SplashScreen {
  width: 100%;
  height: 100%;
}

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

#map {
  height: 100vh;
}
html, body {
  height: 100%;
  margin: 0;
  padding: 0;
}