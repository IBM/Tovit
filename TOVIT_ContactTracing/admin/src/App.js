/* globals alert */
import React, { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import './App.css'
import { fetchCurrentUser } from './helpers/api'
import { Login } from './components/Login'
import { Checkpoints } from './components/Checkpoints'
import { Users } from './components/Users'
import { Account } from './components/Account'

const ViewEnum = {
  checkpoints: 1,
  users: 2,
  account: 3
}

const superPrivilegeLevel = 1
const serverUrl = process.env.REACT_APP_SERVER_DOMAIN

function App () {
  const [isLoggedIn, setIsLoggedIn] = useState(undefined)
  const [currentUser, setCurrentUser] = useState({})
  const [view, setView] = useState(ViewEnum.checkpoints)
  const { t } = useTranslation()
  const hasSuperPrivilege = (currentUser && currentUser.privilege === superPrivilegeLevel)

  const loadCurrentUser = async () => {
    const user = await fetchCurrentUser()
    if (typeof user !== 'undefined') {
      setIsLoggedIn(true)
      setCurrentUser(user)
    }
  }

  useEffect(() => {
    if (typeof isLoggedIn === 'undefined') {
      loadCurrentUser()
    }
  })

  const onSubmitLogin = async (loginSuccessful) => {
    if (loginSuccessful) {
      loadCurrentUser()
    } else {
      alert('Login failed')
    }
  }

  if (isLoggedIn) {
    return (
      <div className='App'>
        <nav class='navbar navbar-dark fixed-top bg-dark flex-md-nowrap p-0 shadow'>
          <a class='navbar-brand col-sm-3 col-md-2 mr-0' href='/admin'>{process.env.REACT_APP_NAME} {t('admin')}</a>
          <ul class='navbar-nav px-3'>
            <li class='nav-item text-nowrap'>
              <a class='nav-link' href={`${serverUrl}/admin/logout`}>{t('logout_button')}</a>
            </li>
          </ul>
        </nav>

        <div class='container-fluid'>
          <div class='row'>
            <nav class='col-md-2 d-none d-md-block bg-light sidebar'>
              <div class='sidebar-sticky'>
                <ul class='nav flex-column'>
                  <li class='nav-item'>
                    <a class='nav-link' onClick={() => setView(ViewEnum.checkpoints)}>
                      {t('menu_checkpoints')}
                    </a>
                  </li>
                  {hasSuperPrivilege && (
                    <li class='nav-item'>
                      <a class='nav-link' onClick={() => setView(ViewEnum.users)}>
                        {t('menu_users')}
                      </a>
                    </li>
                  )}
                  <li class='nav-item'>
                    <a class='nav-link' onClick={() => setView(ViewEnum.account)}>
                      {t('menu_account')}
                    </a>
                  </li>
                </ul>
              </div>
            </nav>

            <main role='main' class='col-md-9 ml-sm-auto col-lg-10 px-4'>
              <p class='mt-3'>{t('welcome')}, {currentUser.username}.</p>
              {view === ViewEnum.checkpoints && (
                <Checkpoints />
              )}
              {hasSuperPrivilege && view === ViewEnum.users && (
                <Users currentUser={currentUser} />
              )}
              {view === ViewEnum.account && (
                <Account />
              )}
            </main>
          </div>
        </div>
      </div>
    )
  } else {
    return (
      <div class='login-container bg-dark text-center text-light'>
        <Login onLoginRequest={onSubmitLogin} />
      </div>
    )
  }
}

export default App
