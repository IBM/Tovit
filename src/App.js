import React, { Fragment } from 'react';
import { Component } from 'react';
import './App.css';
import { NavLink, Switch, Route } from 'react-router-dom';
import mainLogo from'./image/tovit_logo.svg';
import FadeIn from "react-fade-in";
import axios from 'axios'
import renderHTML from 'react-render-html';
import Supplies from './components/Supplies'
import ReactHtmlParser, { processNodes, convertNodeToElement, htmlparser2 } from 'react-html-parser';
import ScriptTag from 'react-script-tag';
import Iframe from './components/Iframe';
import {
  // An object of all themes
  themes,

  // Direct theme values
  white,
  g10,
  g90,
  g100,

  // Specific token values
  interactive01,
  interactive02,
} from '@carbon/themes';

const demoWatson = () => {
  window.loadWatsonAssistantChat({
    integrationID: "", // The ID of this integration.
    region: "eu-gb", // The region your integration is hosted in.
    serviceInstanceID: "", // The ID of your service instance.
    // onLoad: function(instance) { instance.render(); }
    carbonTheme: 'g90'
  }).then(function(instance){
    instance.render();
    instance.updateCSSVariables({
      'BASE-font-family': '"IBM Plex Sans", Helvetica Neue, Arial',
      '$focus': '#3d3d3d',
      '$hover-primary': '#262626',
      '$interactive-01': '#0f6dff',
      'WACLauncher__ButtonContainer': 'black'
    });
  }
  );
}
  const App = () => {
    demoWatson()
    return <div className='app'>
      {/* <h1>Tovit</h1> */}
      <Navigation />
      <Main />

    </div>
  };

  const Navigation = () => (
    <nav>
      <ul>
        <li><NavLink exact activeClassName="current" to='/Dashboard'>Dashboard</NavLink></li>
        <li><NavLink exact activeClassName="current" to='/ContactTracing'>Contact Tracing</NavLink></li>
        <li><NavLink exact activeClassName="current" to='/Supplies'>Supplies</NavLink></li>
        <img src='https://i.imgur.com/pe7PQ04.png'/>
      </ul>
    </nav>
  );

  const Main = () => (
    <Switch>
      <Route exact path='/' component={Home}></Route>
      <Route exact path='/Dashboard' component={Dashboard}></Route>
      <Route exact path='/ContactTracing' component={ContactTracing}></Route>
      <Route exact path='/Supplies' component={Supplies}></Route>
    </Switch>
  );

  const Dashboard = () => {
    return <Iframe source="https://faizhasan123.github.io/COVID-19_Dashboard/"/>
  };

  const Home = () => (
    <img className="SplashScreen" src={mainLogo}/>
  );
  
  
  const ContactTracing = () => {
    return <Iframe source="http://169.50.53.73:3000/"/>
  };



export default App;
