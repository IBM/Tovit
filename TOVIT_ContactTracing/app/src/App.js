import React, { Suspense } from 'react'
import Container from '@material-ui/core/Container'
import IconButton from '@material-ui/core/IconButton'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import BottomNavigation from '@material-ui/core/BottomNavigation'
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction'
import CropFree from '@material-ui/icons/CropFree'
import Face from '@material-ui/icons/Face'
import MenuIcon from '@material-ui/icons/Menu'
import InfoIcon from '@material-ui/icons/Info'
import RoomIcon from '@material-ui/icons/Room'
import PersonIcon from '@material-ui/icons/Person'
import AppBar from '@material-ui/core/AppBar'
import SwipeableDrawer from '@material-ui/core/SwipeableDrawer'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemText from '@material-ui/core/ListItemText'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import CheckpointsPage from './Checkpoints'
import ExposuresPage from './Exposures'
import StatusAlert from './StatusAlert'
import API from './api'
import { Translation } from 'react-i18next'
import MenuItem from '@material-ui/core/MenuItem'
import FormControl from '@material-ui/core/FormControl'
import Select from '@material-ui/core/Select'
import i18n from './i18n'
import supportedLanguages from './languages'

const oneSecond = 1000
const pollingTime = 30 * oneSecond
const checkpointKeyLength = Number(process.env.REACT_APP_CHECKPOINT_KEY_LENGTH)
const adminDomain = process.env.REACT_APP_ADMIN_DOMAIN
const serverDomain = process.env.REACT_APP_SERVER_DOMAIN
const aboutUrl = process.env.REACT_APP_ABOUT_URL

function ListItemLink (props) {
  return <ListItem button component='a' {...props} />
}

class App extends React.Component {
  constructor () {
    super()
    this.state = {
      currentTab: 'checkpoints',
      status: false,
      statusLoaded: false,
      isDrawerOpen: false,
      currentLanguage: i18n.language,
      urlScanState: undefined
    }
  }

  componentDidMount () {
    this.checkUrl().then(() => {
      this.updateStatus()
      setInterval(this.updateStatus.bind(this), pollingTime)
    })
  }

  async checkUrl () {
    const urlParams = new URLSearchParams(window.location.search)
    const checkpointKey = urlParams.get('checkpoint')
    if (checkpointKey) {
      if (checkpointKey.length === checkpointKeyLength) {
        try {
          await API.joinCheckpoint(checkpointKey)
          this.setState({ urlScanState: 'scan-success' })
          window.history.replaceState(null, null, window.location.pathname)
        } catch (e) {
          console.error(e)
          this.setState({ urlScanState: 'scan-error' })
          window.history.replaceState(null, null, window.location.pathname)
        }
      } else {
        this.setState({ urlScanState: 'scan-error' })
        window.history.replaceState(null, null, window.location.pathname)
      }
    }
  }

  async updateStatus () {
    try {
      const exposureStatus = await API.getExposureStatus()
      this.setState({ status: exposureStatus, statusLoaded: true })
    } catch (e) {
      console.error(e)
      this.setState({ status: false, statusLoaded: false })
    }
  }

  resetUrlScanState () {
    this.setState({ urlScanState: undefined })
  }

  onChangeTab (event, newVal) {
    this.setState({ currentTab: newVal })
  }

  openDrawer () {
    this.setState({ isDrawerOpen: true })
  }

  closeDrawer () {
    this.setState({ isDrawerOpen: false })
  }

  onSelectLanguage (event) {
    i18n.changeLanguage(event.target.value)
    this.setState({ currentLanguage: event.target.value })
  }

  render () {
    const { currentTab, status, statusLoaded, isDrawerOpen, currentLanguage, urlScanState } = this.state
    const CurrentPage = (currentTab === 'checkpoints')
      ? CheckpointsPage
      : ExposuresPage

    return (
      <div>
        <AppBar position='static' color='secondary'>
          <Container maxWidth='sm' style={{ flexGrow: 1 }}>
            <Toolbar>
              <IconButton
                edge='start'
                color='inherit'
                aria-label='open drawer'
                onClick={this.openDrawer.bind(this)}
              >
                <MenuIcon />
              </IconButton>
              <Typography variant='h6' component='h1' style={{ flexGrow: 1 }}>
                {process.env.REACT_APP_NAME}
              </Typography>
            </Toolbar>
          </Container>
        </AppBar>
        <Container maxWidth='sm' style={{ marginBottom: 76 }}>
          <Suspense fallback='loading'>
            <StatusAlert status={status} onExposuresTab={currentTab === 'status'} />
            <CurrentPage status={status} statusLoaded={statusLoaded} urlScanState={urlScanState} resetUrlScanState={this.resetUrlScanState.bind(this)} />
            {supportedLanguages.length > 1 && (
              <Container style={{ textAlign: 'center' }}>
                <div>
                  <FormControl style={{ marginTop: '20px' }}>
                    <Select
                      labelId='language-select-label'
                      id='language-select'
                      value={currentLanguage}
                      onChange={this.onSelectLanguage.bind(this)}
                    >
                      { supportedLanguages.map(language => {
                        return (
                          <MenuItem value={language.id}>{language.name}</MenuItem>
                        )
                      }) }
                    </Select>
                  </FormControl>
                </div>
              </Container>
            )}
          </Suspense>
        </Container>
        <BottomNavigation
          value={currentTab}
          style={{
            width: '100%',
            position: 'fixed',
            bottom: 0
          }}
          onChange={this.onChangeTab.bind(this)}
          showLabels
        >
          <BottomNavigationAction label={<Translation>{t => t('checkpointsTab')}</Translation>} value='checkpoints' icon={<CropFree />} />
          <BottomNavigationAction label=<Translation>{t => t('statusTab')}</Translation> value='status' icon={<Face />} />
        </BottomNavigation>
        <SwipeableDrawer
          open={isDrawerOpen}
          onOpen={() => {}}
          onClose={this.closeDrawer.bind(this)}
        >
          <List component='nav' aria-label='settings'>
            <ListItemLink style={{ width: 250 }} href={aboutUrl} target='_blank'>
              <ListItemIcon>
                <InfoIcon />
              </ListItemIcon>
              <ListItemText primary=<Translation>{t => t('menuAboutButton')}</Translation> />
            </ListItemLink>
            <ListItemLink style={{ width: 250 }} href={`${serverDomain}/checkpoint`} target='_blank'>
              <ListItemIcon>
                <RoomIcon />
              </ListItemIcon>
              <ListItemText primary=<Translation>{t => t('menuCheckpointButton')}</Translation> />
            </ListItemLink>
            <ListItemLink style={{ width: 250 }} href={`${adminDomain}/admin`} target='_blank'>
              <ListItemIcon>
                <PersonIcon />
              </ListItemIcon>
              <ListItemText primary=<Translation>{t => t('menuAdminButton')}</Translation> />
            </ListItemLink>
          </List>
        </SwipeableDrawer>
      </div>
    )
  }
}

export default App
