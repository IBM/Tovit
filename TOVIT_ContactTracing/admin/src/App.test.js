/* globals test, expect */
import React from 'react'
import App from '../src/App'
import TestRenderer from 'react-test-renderer'
import { I18nextProvider } from 'react-i18next'
import i18n from './i18n'

test('App component', () => {
  const component = TestRenderer.create(
    <I18nextProvider i18n={i18n}>
      <App />
    </I18nextProvider>)
  expect(component).toMatchSnapshot()
})
