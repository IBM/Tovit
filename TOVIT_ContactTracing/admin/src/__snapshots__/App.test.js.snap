// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component 1`] = `
<div
  class="login-container bg-dark text-center text-light"
>
  <form
    class="form-signin"
    onSubmit={[Function]}
  >
    <h1
      class="h3 mb-3 font-weight-normal"
    >
      Please sign in
    </h1>
    <label
      class="sr-only"
      for="username"
    >
      Username
      :
    </label>
    <br />
    <input
      class="form-control"
      id="username"
      name="username"
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value=""
    />
    <br />
    <label
      class="sr-only"
      for="password"
    >
      Password
      :
    </label>
    <br />
    <input
      class="form-control"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <br />
    <br />
    <button
      class="btn btn-lg btn-warning btn-block"
      type="submit"
    >
      Sign in
    </button>
    <p
      class="mt-5 mb-3 text-muted"
    >
      Don't have an account? You can request access
       
      <a
        class="text-warning"
        href="http://my-form.my-org"
        target="_blank"
      >
        here
      </a>
      .
    </p>
    <p
      class="mt-5 mb-3 text-muted"
    >
      TrackCOVID
       
      Admin
    </p>
  </form>
</div>
`;
